File: targets/java-web/src/main/java/com/example/vulnapp/vulns/Vuln13Hash.java
package com.example.vulnapp.vulns;

import com.example.vulnapp.vulns.crypto.WeakHash;
import org.springframework.web.bind.annotation.*;

@RestController
public class Vuln13Hash {
    @GetMapping("/vuln13/hash")
    public String hash(@RequestParam String p) {
        // weak-hash sink lives in WeakHash.md5 (VULN-13); this endpoint exercises it
        return WeakHash.md5(p);
    }
}

File: targets/kotlin-web/src/main/kotlin/com/example/vulnapp/vulns/Vuln13Hash.kt
package com.example.vulnapp.vulns

import org.springframework.web.bind.annotation.*


@RestController
class Vuln13Hash {

    @GetMapping("/vuln13/hash")
    fun hash(@RequestParam p: String): String = com.example.vulnapp.vulns.crypto.WeakHash.md5(p)
}
